Packages/io.chainsafe.web3-unity/Runtime/Scripts/EVM/Remote/CSServer.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Threading.Tasks;
+using ChainSafe.Gaming.UnityPackage;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace Scripts.EVM.Remote
+{
+    public class CSServer
+    {
+        private static readonly string host = "https://api.gaming.chainsafe.io/v1/projects/";
+
+        /// <summary>
+        /// Unity web request helper function to retrieve data.
+        /// </summary>
+        /// <param name="_path"></param>
+        /// <returns></returns>
+        public static async Task<string> GetData(string _path)
+        {
+            using UnityWebRequest webRequest = UnityWebRequest.Get($"{host}{Web3Accessor.Web3.ProjectConfig.ProjectId}{_path}");
+            await webRequest.SendWebRequest();
+            if (webRequest.result != UnityWebRequest.Result.Success)
+            {
+                Debug.LogError("Error: Your project ID doesn't have a marketplace, please go to dashboard and create items " + webRequest.error);
+                return null;
+            }
+            var json = webRequest.downloadHandler.text;
+            return json;
```

Comment:
You can instead return a Deserialized data T with the method signature changing to `public static async Task<T> GetData<T>(string _path)`

Reply:
this way you don't have to deserialize everywhere :)

Reply:
done